clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -319,27 +330,50 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
         return committed(partitions, Duration.ofMillis(defaultApiTimeoutMs));
     }
 
+    /**
+     * Retrieve the last committed offset for the given partition (whether the commit happened by this process or
+     * another). This offset will be used as the position for the consumer in the event of a failure.
+     * <p>
+     * If the timeout specified by {@code default.api.timeout.ms} expires
+     * {@link org.apache.kafka.common.errors.TimeoutException} is thrown.
+     *
+     * @param partitions The partition to check
+     * @param timeout The maximum time to block.
+     * @return The last committed offset and metadata or null if there was no prior commit
+     * @throws org.apache.kafka.common.errors.WakeupException if {@link #wakeup()} is called before or while this
+     *             function is called
+     * @throws org.apache.kafka.common.errors.InterruptException if the calling thread is interrupted before or while
+     *             this function is called
+     * @throws org.apache.kafka.common.errors.AuthenticationException if authentication fails. See the exception for more details
+     * @throws org.apache.kafka.common.errors.AuthorizationException if not authorized to the topic or to the
+     *             configured groupId. See the exception for more details
+     * @throws org.apache.kafka.common.KafkaException for any other unrecoverable errors
+     * @throws org.apache.kafka.common.errors.TimeoutException if the committed offset cannot be found before
+     *             the timeout specified by {@code default.api.timeout.ms} expires.
+     */
     @Override
     public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition> partitions,
                                                             final Duration timeout) {
+        maybeWakeup();
         maybeThrowInvalidGroupIdException();
+
         if (partitions.isEmpty()) {
             return new HashMap<>();
         }
 
         final OffsetFetchApplicationEvent event = new OffsetFetchApplicationEvent(partitions);
+        ensureNoInflightBlockingFutures(event.future());
```

Comment:
it's a bit of an awkward place, but we want to make sure when we set an active future, there isn't one at the moment.